src/StreamingBodyParser/FormUrlencodedParser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitter;
+use Psr\Http\Message\RequestInterface;
+use React\Http\HttpBodyStream;
+
+final class FormUrlencodedParser extends EventEmitter
+{
+    /**
+     * @var RequestInterface
+     */
+    private $request;
+
+    /**
+     * @var HttpBodyStream
+     *
+     * @internal
+     */
+    public $body;
+
+    /**
+     * @var string
+     *
+     * @internal
+     */
+    public $buffer = '';
+
+    /**
+     * @param RequestInterface $request
+     */
+    public function __construct(RequestInterface $request)
+    {
+        $this->request = $request;
+        $this->body = $this->request->getBody();
```

Comment:
WDYT? Since the class isn't internal, it isn't guaranteed that `$this->body` is an instance of `HttpBodyStream`.